app/frontend/src/search/hits.js

```diff
@@ -0,0 +1,122 @@
+/* global window */
+import { extractQueryParams } from '../lib/utils';
+import '../polyfill/remove.polyfill';
+
+export const renderContent = (renderOptions) => {
+  const { results, widgetParams } = renderOptions;
+
+  if (results) {
+    widgetParams.container.innerHTML = addHeadingMarkup(results.nbHits);
+    if (results.query.length >= widgetParams.threshold) {
+      addJobAlertMarkup(widgetParams.alert);
+      document.querySelector('#job-alert-link').href = getJobAlertLink(window.location.href);
+    } else {
+      removeJobAlertMarkup();
+    }
+    hideServerMarkup();
+  }
+};
+
+export const addJobAlertMarkup = container => {
+  !document.getElementById('job-alert-cta') ? container.insertAdjacentHTML('beforeend', customTemplates.alert) : false; 
+};
+
+export const removeJobAlertMarkup = () => {
+  const el = document.getElementById('job-alert-cta');
+  return el && el.remove(); 
+};
+
+const JOB_ALERT_URL = '/subscriptions/new';
+
+export const getJobAlertLink = url => {
+  const paramsObj = extractQueryParams(url, ['keyword', 'location', 'radius']);
+  let queryString = '';
+
+  Object.keys(paramsObj).map(key => {
+    queryString += getJobAlertLinkParam(key, paramsObj[key]);
+
+    if (key === 'location') {
+      queryString += getJobAlertLinkParam('location_category', paramsObj[key]);
+    }
+  });
+
+  return `${JOB_ALERT_URL}?${queryString}`;
+};
+
+export const getJobAlertLinkParam = (key, value) => {
+  return encodeURIComponent(`search_criteria[${key}]`) + `=${value.replace(' ', '+')}&`;
+};
+
+export const addHeadingMarkup = (numberHits) => {
+  const keyword = document.getElementById('keyword').value;
+  const location = document.getElementById('location').value;
+  const prefix = keyword || location ? ' match ' : ' listed';
+  const postfix = `${prefix}${createCapitalisedStringWithPrefix('', keyword)} ${createCapitalisedStringWithPrefix('near', location)}`;
+  const hits = keyword || location ? numberHits : `There are ${numberHits}`;
+
+  return `${hits} jobs ${postfix}`;
+};
+
+export const createCapitalisedStringWithPrefix = (pre, string) => {
+  return string ? `${pre} ${string.replace(/\b\w/g, l => l.toUpperCase() )}` : '';
+};
+
+export const hideServerMarkup = () => {
+  document.querySelector('ul.vacancies').style.display = 'none';
+
+  if (document.querySelector('ul.pagination-server')) {
+    document.querySelector('ul.pagination-server').style.display = 'none';
+  }
+};
+
+export const snakeCaseToHumanReadable = value => value.toLowerCase().replace(/_/g, ' ');
```

Comment:
I think this is to blame for the following regression: we are losing the working patterns punctuation. Should not be lower case. Sorry, should have spotted this before.

<img width="566" alt="Screenshot 2020-06-01 at 12 57 25" src="https://user-images.githubusercontent.com/60350599/83407850-77cff600-a409-11ea-8b2e-375c3556d9f0.png">


<img width="566" alt="Screenshot 2020-06-01 at 12 55 40" src="https://user-images.githubusercontent.com/60350599/83407848-769ec900-a409-11ea-8dd5-b54a3f88b82e.png">

Reply:
Normally handled by this method in vacancy_presenter.rb:

```ruby
def working_patterns
    return unless working_patterns?

    patterns = model_working_patterns.map do |working_pattern|
      Vacancy.human_attribute_name("working_patterns.#{working_pattern}").downcase
    end.join(', ')

    I18n.t("jobs.working_patterns_info_#{model_working_patterns.count > 1 ? 'many' : 'one'}", patterns: patterns)
        .capitalize
  end
```